### DISCUSSION TITLE
Redirecting audio output to WAV file

### DISCUSSION THREAD

**Discussion by hash6iron** [pschatzmann/arduino-audio-tools#965] [Q&A] [answered]
Hello,

Congratulations for you excellent work with this library. I'm coding a device that play TAP and TZX files for ZX Spectrum computer using a ESP32 Audiokit module. You can know about it in my GitHub https://github.com/hash6iron/powadcr

I'd like to taka advantage the output that is generate with square waves to create a wav file in the SD to debug operations but I don't know how to do it. Could you send me an example to try it?

On the other hand. I've noticed when I change volume the stream stop, I'm generating the stream on real time , I create a wave and write it in a buffer to output. How can I change volume without sounds cut?

Thanks in advance.

**Comment by pschatzmann** [comment, marked as answer]
I dont't quite understand your problem: just read the Wiki Documentation about Encoding and replace the NoiseGenerator with a SquareWaveGenerator. 

There is a wiki about volume, but in this case you can just set the amplitude directly.

Your statement about using a buffer does not make any sense to me.

I suggest to spend some more time reading the availalble documentation which includes the wiki and class documentation

**Comment by hash6iron** [comment]
Hello,

Thanks for your quick response.

I have written a specific square wave generator adapted to my needed, but
it's similar at some wave generator example. Then I'd like to redirect this
example Sine wave generator to WAV file (instead of to speaker)

I have read a lot of possible but I don't finish to understand.

I appreciate your help
Thanks.

**Comment by pschatzmann** [comment]
Did you read the chapter about encoding ?
What part of the given file example did you not understand ?

**Comment by hash6iron** [comment]
Not I don't any doubt about the example but I don't know how to redirect
for example Sine wave generated to WAV file instead of line out. Anyway I'm
going to review again this chapter if you think that the information is
perfectly clear.

Best regards.

**Reply by pschatzmann** [reply]
But the given example does exactly what you are asking for! 
It even gives you some valuable hints...

**Comment by hash6iron** [comment]
Ok.

I'm going to review again.

Thanks so much.
BR.

**Comment by hash6iron** [comment]
Hi Phil,

Another question

Is it possible to know the battery level with Audiokit (ES8388 audio
processor version)?

Thanks in advance.
